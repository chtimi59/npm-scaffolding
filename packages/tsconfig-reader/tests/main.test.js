const lib = require('../index.js')

const base = __dirname.replace(/\\/g, '\\\\')
const removeAbs = s => s.replace(new RegExp(base, 'g'), "<ABSOLUTE>")

describe('tsconfig-reader', function () {
    it('json', function () {
        const tsconfig = new lib(`${__dirname}/tsconfig.json`)
        const result = removeAbs(JSON.stringify(tsconfig.json, null, 4).replace(/\\\\/g, '\\'))
        expect(result).toMatchSnapshot()
    })
    it('alias', function () {
        const tsconfig = new lib(`${__dirname}/tsconfig.json`)
        const result = removeAbs(tsconfig.alias('@hello/something'))
        expect(result).toMatchSnapshot()
    })
})


