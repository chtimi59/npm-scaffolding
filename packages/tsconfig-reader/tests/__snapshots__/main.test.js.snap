// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tsconfig-reader alias 1`] = `"<ABSOLUTE>\\\\src\\\\bar\\\\something"`;

exports[`tsconfig-reader json 1`] = `
"{
    \\"compilerOptions\\": {
        \\"module\\": \\"commonjs\\",
        \\"target\\": \\"es5\\",
        \\"lib\\": [
            \\"es2017\\",
            \\"dom\\"
        ],
        \\"noImplicitAny\\": true,
        \\"noImplicitThis\\": false,
        \\"noUnusedLocals\\": true,
        \\"sourceMap\\": true,
        \\"outDir\\": \\"<ABSOLUTE>\\\\folder\\\\tsOut\\",
        \\"strict\\": true,
        \\"declaration\\": true,
        \\"jsx\\": \\"react\\",
        \\"moduleResolution\\": \\"node\\",
        \\"baseUrl\\": \\"<ABSOLUTE>\\",
        \\"paths\\": {
            \\"@hello/*\\": [
                \\"<ABSOLUTE>\\\\src\\\\bar\\\\*\\"
            ]
        },
        \\"declarationDir\\": \\"<ABSOLUTE>\\\\folder\\\\tsOut\\"
    },
    \\"include\\": [
        \\"<ABSOLUTE>\\\\src\\\\**\\\\*\\"
    ],
    \\"tsconfig\\": {
        \\"extends\\": \\"<ABSOLUTE>\\\\folder\\\\tsconfig.json\\",
        \\"compilerOptions\\": {
            \\"outDir\\": \\"<ABSOLUTE>\\\\dist\\",
            \\"declaration\\": false,
            \\"declarationDir\\": \\"<ABSOLUTE>\\\\dist\\"
        }
    }
}"
`;
